frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,439 @@
+        let active_era = <ActiveEra<T>>::mutate(|active_era| {
+            let new_index = active_era.as_ref().map(|info| info.index + 1).unwrap_or(0);
+            *active_era = Some(ActiveEraInfo {
+                index: new_index,
+                // Set new active era start in next `on_finalize`. To guarantee usage of `Time`
+                start: None,
+            });
+            new_index
+        });
+
+        // let bonding_duration = T::BondingDuration::get();
+
+        BondedEras::mutate(|bonded| {
+            bonded.push((active_era, start_session));
+
+            // if active_era > bonding_duration {
+            //     let first_kept = active_era - bonding_duration;
+            //
+            //     // prune out everything that's from before the first-kept index.
+            //     let n_to_prune = bonded.iter()
+            //         .take_while(|&&(era_idx, _)| era_idx < first_kept)
+            //         .count();
+            //
+            //     // kill slashing metadata.
+            //     for (pruned_era, _) in bonded.drain(..n_to_prune) {
+            //         slashing::clear_era_metadata::<T>(pruned_era);
+            //     }
+            //
+            //     if let Some(&(_, first_session)) = bonded.first() {
+            //         T::SessionInterface::prune_historical_up_to(first_session);
+            //     }
+            // }
+        });
+    }
+
+    /// Compute payout for era.
+    fn end_era(active_era: ActiveEraInfo<MomentOf<T>>, _session_index: SessionIndex) {
+        // Note: active_era_start can be None if end era is called during genesis config.
+        if let Some(active_era_start) = active_era.start {
+            // The set of total amount of staking.
+            let now = T::Time::now();
+            let era_duration = now - active_era_start;
+
+            if !era_duration.is_zero() {
+                let total_payout = T::Currency::total_issuance();
+                let for_dapps = T::GetForDappsStaking::get_era_staking_amount(active_era.index);
+                let for_security =
+                    T::GetForSecurityStaking::get_era_staking_amount(active_era.index);
+
+                let (for_security_reward, for_dapps_rewards) =
+                    T::ComputeTotalPayout::compute_total_payout(
+                        total_payout,
+                        era_duration.saturated_into::<u64>(),
+                        for_security,
+                        for_dapps,
+                    );
+
+                <ForSecurityEraReward<T>>::insert(active_era.index, for_security_reward);
+                <ForDappsEraReward<T>>::insert(active_era.index, for_dapps_rewards);
+            }
+        }
+    }
+
+    /// Plan a new era. Return the potential new staking set.
+    fn new_era(start_session_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+        // Increment or set current era.
+        let current_era = CurrentEra::mutate(|s| {
```

Comment:
I guess it's pretty much code for simple increment =)

What's about:
```rust
let current_era = CurrentEra::get().map(|s| s + 1).unwrap_or(0);
CurrentEra::put(Some(current_era)); 
```

Reply:
In some code on Substrate, mutate is used to get and change the value from storage.
I need to confirm the reason, but this is the recommended style in the Substrate's main repository.
https://github.com/paritytech/substrate/blob/master/frame/staking/src/lib.rs#L2372

Reply:
And Jam says on the riot:
> I think the main benefit is that mutate enforces a single read and write, so as long as you only have a single mutate per storage item, you will be doing things as efficiently as possible. Otherwise you might accidentally read from or write to storage multiple times. 

https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$158559169115385wPbmc:matrix.parity.io?via=matrix.parity.io&via=matrix.org&via=web3.foundation

Reply:
But, in that case you also has single read and write =) didnt?

Reply:
So, it doesn't make a big sense, let's finally merge it